geonode/base/views.py

```diff
@@ -43,6 +43,72 @@
 from geonode.base.forms import BatchEditForm, OwnerRightsRequestForm
 from geonode.base.forms import CuratedThumbnailForm
 from geonode.notifications_helper import send_notification
+from geonode.base.forms import UserAndGroupPermissionsForm
+from django.contrib.auth import get_user_model
+from geonode.tasks.tasks import set_permissions
+from django.contrib.auth.models import Group
+
+
+def user_and_group_permission(request, model):
+    if not request.user.is_superuser:
+        raise PermissionDenied
+
+    model_mapper = {
+        "profile": get_user_model(),
+        "group": Group
+    }
+
+    model_class = model_mapper[model]
+
+    ids = request.POST.get("ids")
+    if "cancel" in request.POST or not ids:
+        return HttpResponseRedirect(
+            '/admin/{}/{}/'.format(model_class._meta.app_label, model)
+        )
+
+    if request.method == 'POST':
+        form = UserAndGroupPermissionsForm(request.POST)
+        ids = ids.split(",")
+        if form.is_valid():
+            resources_names = [layer.name for layer in form.cleaned_data.get('layers')]
+            users_usernames = [user.username for user in model_class.objects.filter(
+                id__in=ids)] if model == 'profile' else None
+            groups_names = [group.name for group in model_class.objects.filter(
+                id__in=ids)] if model == 'group' else None
+
+            if users_usernames and 'AnonymousUser' in users_usernames and \
+                    (not groups_names or 'anonymous' not in groups_names):
```

Comment:
would be nice to wrap it in function with an informative name

Reply:
@petrus7, nice catch, I'll make changes to put this in a function.